build-tools/packages/build-cli/src/commands/generate/entrypoints.ts

```diff
@@ -3,6 +3,135 @@
+import type { Package } from "@fluidframework/build-tools";
+import { Flags } from "@oclif/core";
+import { PackageCommand } from "../../BasePackageCommand.js";
+import type { PackageSelectionDefault } from "../../flags.js";
+import {
+	generateEntrypoints,
+	generateNode10TypeEntrypoints,
+	getOutputConfiguration,
+	optionDefaults,
+} from "../../library/commands/generateEntrypoints.js";
+import { ApiLevel, unscopedPackageNameString } from "../../library/index.js";
 
-export default class GenerateEntrypointsCommand_ extends GenerateEntrypointsCommand {}
+/**
+ * Generates type declarations files for Fluid Framework APIs to support API levels (/alpha, /beta. etc.).
+ */
+export default class GenerateEntrypointsCommand extends PackageCommand<
+	typeof GenerateEntrypointsCommand
+> {
+	static readonly description =
+		`Generates type declaration entrypoints for Fluid Framework API levels (/alpha, /beta. etc.) as found in package.json "exports"`;
+
+	static readonly flags = {
+		mainEntrypoint: Flags.file({
+			description: "Main entrypoint file containing all untrimmed exports.",
+			default: optionDefaults.mainEntrypoint,
+			exists: true,
+		}),
+		outDir: Flags.directory({
+			description: "Directory to emit entrypoint declaration files.",
+			default: optionDefaults.outDir,
+			exists: true,
+		}),
+		outFilePrefix: Flags.string({
+			description: `File name prefix for emitting entrypoint declaration files. Pattern of '${unscopedPackageNameString}' within value will be replaced with the unscoped name of this package.`,
+			default: optionDefaults.outFilePrefix,
+		}),
+		outFileAlpha: Flags.string({
+			description: "Base file name for alpha entrypoint declaration files.",
+			default: optionDefaults.outFileAlpha,
+		}),
+		outFileBeta: Flags.string({
+			description: "Base file name for beta entrypoint declaration files.",
+			default: optionDefaults.outFileBeta,
+		}),
+		outFileLegacy: Flags.string({
+			description: "Base file name for legacy entrypoint declaration files.",
+			default: optionDefaults.outFileLegacy,
+		}),
+		outFilePublic: Flags.string({
+			description: "Base file name for public entrypoint declaration files.",
+			default: optionDefaults.outFilePublic,
+		}),
+		outFileSuffix: Flags.string({
+			description:
+				"File name suffix including extension for emitting entrypoint declaration files.",
+			default: optionDefaults.outFileSuffix,
+		}),
+		node10TypeCompat: Flags.boolean({
+			description: `Optional generation of Node10 resolution compatible type entrypoints matching others.`,
+		}),
+		...PackageCommand.flags,
+	};
+
+	// If no selection flags are used, select the current directory. (`--dir .`)
+	protected defaultSelection = "dir" as PackageSelectionDefault;
+
+	/**
+	 * This is called for every package selected/filtered by the selection arguments.
+	 */
+	protected async processPackage(pkg: Package): Promise<void> {
+		const { mainEntrypoint, node10TypeCompat } = this.flags;
+		const { packageJson } = pkg;
```

Comment:
These are the most relevant changes. The contents of the "run" function is now in processPackage. The default selection is "dir" so that existing commands will continue to work.